crates/winner-selection/src/solution.rs

```diff
@@ -0,0 +1,150 @@
+//! Minimal solution and order data structures.
+//!
+//! These structs contain only the data needed for winner selection,
+//! making them small enough to efficiently send to/from the Pod Service.
+
+pub use state::{RankType, Unscored};
+use {
+    crate::{
+        primitives::{OrderUid, Side},
+        state,
+    },
+    alloy::primitives::{Address, U256},
+    std::collections::HashMap,
+};
+pub type Scored = state::Scored<U256>;
+pub type Ranked = state::Ranked<U256>;
+
+/// Minimal solution data needed for winner selection.
+///
+/// This contains only what's absolutely necessary to run the winner selection
+/// algorithm. Autopilot and driver convert their full solution types to this
+/// minimal format before sending to the Pod Service.
+///
+/// Estimated size: ~1.7KB for a solution with 5 orders and 10 unique tokens.
+#[derive(Debug, Clone)]
+pub struct Solution<State> {
+    /// Solution ID from solver (unique per solver).
+    pub id: u64,
+
+    /// Solver's submission address (used for identifying the solver).
+    pub solver: Address,
+
+    /// Orders executed in this solution.
+    ///
+    /// Uses Vec instead of HashMap for smaller serialization size.
+    pub orders: Vec<Order>,
+
+    /// Uniform clearing prices for all tokens in the solution.
+    ///
+    /// Maps token address to its price in the native token.
+    /// These are the prices at which all orders trading these tokens are
+    /// settled.
+    pub prices: HashMap<Address, U256>,
+
+    /// State marker (score and ranking information).
+    state: State,
+}
+
+impl<T> Solution<T> {
+    /// Get the solution ID.
+    pub fn id(&self) -> u64 {
```

Comment:
Either we should drop these getter functions or make the fields private. Having both doesn't make much sense.